src/engine/routing_algorithms/many_to_many.cpp

```diff
@@ -179,11 +184,15 @@ void relaxOutgoingEdges(const DataFacade<mld::Algorithm> &facade,
     BOOST_ASSERT(!facade.ExcludeNode(node));
 
     const auto &partition = facade.GetMultiLevelPartition();
-    const auto &cells = facade.GetCellStorage();
-    const auto &metric = facade.GetCellMetric();
 
-    const auto level = getNodeQueryLevel(partition, node, args...);
+    const auto level = getNodeQueryLevel<STOP_AT_TOP_LEVEL>(partition, node, args...);
+
+    // Break outgoing edges relaxation if node at the restricted level
+    if (level == INVALID_LEVEL_ID)
+        return;
 
+    const auto &cells = facade.GetCellStorage();
```

Comment:
These two don't come up later in this PR. Why do we need them now?

Reply:
ahh. sorry. Didn't see them in the diff from before. I thought you were adding new unused variables here. fine to keep as is.